src/lambda.rs

```diff
@@ -0,0 +1,220 @@
+/// * Processes environment variables and makes them available to your handler
+/// * Initialises a shared context object, which is passed to your handler.
+/// * Deserialises a batch of messages and passes each one to your handler.
+///
+/// ## Writing a message handler
+///
+/// To write a message handler, you need to define four elements:
+///
+/// * The `Message` structure, which defines the structure of the messages which will be sent to the SQS
+///    queue, and then forwarded on to your Lambda.
+/// * The `Env` structure, which defines the expected environment variables your Lambda will receive.
+/// * The `Context` structure, which is provided the `Env` structure, and represents the shared state
+///   that will be passed into your message handler. This structure needs to implement the [LambdaContext] trait.
+/// * The `message_handler` function, which accepts a `Message` and a `Context`, and performs the desired actions.
+///
+/// # Example
+///
+/// ```no_run
+/// use anyhow::Result;
+/// use async_trait::async_trait;
+/// use clap::Parser;
+/// use serde::Deserialize;
+/// use std::fmt::Debug;
+/// use std::sync::Arc;
+///
+/// use cobalt_aws::lambda::{run_message_handler, Error, LambdaContext};
+///
+/// #[tokio::main]
+/// async fn main() -> Result<(), Error> {
+///     run_message_handler(message_handler).await
+/// }
+///
+/// /// The structure of the messages we expect to see on the queue.
+/// #[derive(Debug, Deserialize)]
+/// pub struct Message {
+///     pub target: String,
+/// }
+///
+/// /// Configuration we receive from environment variables.
+/// ///
+/// /// Note: all fields should be tagged with the `#[clap(env)]` attribute.
+/// #[derive(Debug, Parser)]
+/// pub struct Env {
+///     #[clap(env)]
+///     pub greeting: String,
+/// }
+///
+/// /// Shared context we build up before invoking the lambda runner.
+/// #[derive(Debug)]
+/// pub struct Context {
+///     pub greeting: String,
+/// }
+///
+/// #[async_trait]
+/// impl LambdaContext<Env> for Context {
+///     /// Initialise a shared context object from which will be
+///     /// passed to all instances of the message handler.
+///     async fn from_env(env: &Env) -> Result<Context> {
+///         Ok(Context {
+///             greeting: env.greeting.clone(),
+///         })
+///     }
+/// }
+///
+/// /// Process a single message from the SQS queue, within the given context.
+/// async fn message_handler(message: Message, context: Arc<Context>) -> Result<()> {
+///     tracing::debug!("Message: {:?}", message);
+///     tracing::debug!("Context: {:?}", context);
+///
+///     // Log a greeting to the target
+///     tracing::info!("{}, {}!", context.greeting, message.target);
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// # Error handling
+///
+/// If any errors are raised during init, or from the `message_handler` function, then the entire message
+/// batch will be considered to have failed. Error messages will be logged in CloudWatch, and the message batch
```

Comment:
We don't specifically log anything to CloudWatch, and it's not clear to me if there are things the user could do to consume the errors with something other than CloudWatch. Would something along the lines of "Error messages will be logged to stderr in a format suitable for consumption by CloudWatch" be more useful here?

Reply:
Very good point, this is a subtle but probably important distinction.